collection/lens.rkt

```diff
@@ -39,6 +67,21 @@
   (make-lens (λ (target) (nth target index))
              (λ (target value) (set-nth target index value))))
 
+(define (mapper-lens lens)
```

Comment:
I think `map-lens` is the right name. It keeps things consistent, and it keeps the same paradigm of lens signatures looking like partially-applied versions of their equivalent functions.